### DISCUSSION TITLE
Refresh shared dashboard on demand

### DISCUSSION THREAD

**Discussion by pauljura** [getredash/redash#7124] [Ideas]
Hi folks, we have a few shared dashboards that are looked at maybe a few times per day. What I've noticed is that quite often the data is last refresh several days ago, which makes it not very useful. I would like for the data to be fairly fresh when the dashboard is viewed. But I don't want to set it to auto-refresh every minute because that's putting unnecessary load on the database. I can think of some possible solutions:
* add a "refresh" button on the dashboard so viewers can manually trigger a refresh
* add a setting that automatically refreshes if the dashboard is more than X minutes old when viewed (configurable)
* or a combination: add a manual refresh button but it's only allowed when the data is more than X minutes old (to prevent abuse and possible overload)

Thanks. Keep up the good work.

**Comment by Gordory** [comment]
I can relate to that! Common needed feature to use public dashboards without manually checking them

**Comment by cliuds** [comment]
I agree! We are running into an issue where dashboards with parameters are not refreshing - having at least a manual refresh button would allow folks to update the data of the public dashboards. Right now, it's the most frustrating part of Redash dashboards we are trying to debug to no avail.

**Reply by Jengro777** [reply]
```
import os
import psycopg2
import requests

pg_host = '127.0.0.1'
pg_user = 'postgres'
pg_pass = '***********'  //your pwd
pg_db = 'postgres'

def psql():
    try:
        conn = psycopg2.connect(
            host=pg_host,
            user=pg_user,
            password=pg_pass,
            dbname=pg_db
        )

        cur = conn.cursor()


        cur.execute("select id from queries;")

        rows = cur.fetchall()

  
        for row in rows:
            id = row[0]
            refresh_remmner(id,"ghana","tospinomall")
            refresh_data(id,"tospinomall","gh")
            refresh_data(id,"gh2b","gh")


       
        conn.commit()

        print('------------ start -----------------')

    except Exception as e:
        print(f"An error occurred: {e}")
    finally:
      
        if cur:
            cur.close()
        if conn:
            conn.close()




def refresh_data(id,site,DataSource):

    url = "https://127.0.0.1:5000/api/queries/"+str(id)+"/results"

    payload = {
        "parameters": {
            "site": site,
            "DataSource": DataSource
        },
        "max_age": 1
    }
    headers = {
    "Authorization": "*********",  //your key
    "content-type": "application/json"
    }

    response = requests.post(url, json=payload, headers=headers)

    print(response.json())



def refresh_remmner(id,country,site):

    url = "https://127.0.0.1:5000/api/queries/"+str(id)+"/results"

    payload = {
        "parameters": {
            "country": country,
            "site": site
        },
        "max_age": 1
    }
    headers = {
    "Authorization": "***********",  //your key
    "content-type": "application/json"
    }

    response = requests.post(url, json=payload, headers=headers)

    print(response.json())





psql()
print('------------ end -----------------')
```

Can you create a scheduled task to update query data on a regular basis
